### PR TITLE
XYChart: Improved new tooltip

### LINKED ISSUE #75688: Tooltip: Improved XYChart tooltip
[Figma link](https://www.figma.com/file/tW8PMdzLb7hYYyBpb2xaYR/Universal-Tooltip?node-id=238%3A3959&mode=dev)

TODO
- [ ] `XYChartTooltip` component to use the new info
- [ ] Header
- [ ] Content
- [ ] Footer
- [ ] Clean up
- [ ] Remove old tooltip component

### PR DESCRIPTION
**Pull request by adela-almasan**
Moved out of #70872

xy

Fixes #75688

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/xychart/XYChartPanel.tsx [hunk 1]
```diff
@@ -184,11 +186,11 @@ export const XYChartPanel2 = (props: Props) => {
...
 
     const legendStyle = {
-      flexStart: css`
-        div {
-          justify-content: flex-start !important;
-        }
-      `,
+      flexStart: css({
+        div: {
+          justifyContent: 'flex-start !important',
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/XYChartPanel.tsx:191]: does this !important work here? I had issues before with getting this to work with object css style 🤔

File: public/app/plugins/panel/xychart/XYChartPanel.tsx [hunk 2]
```diff
@@ -215,47 +217,68 @@ export const XYChartPanel2 = (props: Props) => {
     <>
       <VizLayout width={props.width} height={props.height} legend={renderLegend()}>
         {(vizWidth: number, vizHeight: number) => (
-          <UPlotChart config={builder} data={facets} width={vizWidth} height={vizHeight} />
+          <UPlotChart config={builder} data={facets} width={vizWidth} height={vizHeight}>
+            {config.featureToggles.newVizTooltips && props.options.tooltip.mode !== TooltipDisplayMode.None && (
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/XYChartPanel.tsx:221]: Should we also wrap the `UPlotChart` in the FF check / options check? does it make sense to render empty UPlotChart here?

File: public/app/plugins/panel/xychart/module.tsx [hunk 3]
```diff
@@ -3,11 +3,11 @@ import { commonOptionsBuilder } from '@grafana/ui';
 
 import { AutoEditor } from './AutoEditor';
 import { ManualEditor } from './ManualEditor';
-import { XYChartPanel2 } from './XYChartPanel2';
+import { XYChartPanel } from './XYChartPanel';
 import { getScatterFieldConfig } from './config';
 import { Options, FieldConfig, defaultFieldConfig } from './panelcfg.gen';
 
-export const plugin = new PanelPlugin<Options, FieldConfig>(XYChartPanel2)
+export const plugin = new PanelPlugin<Options, FieldConfig>(XYChartPanel)
```
Comment (nmarrs) [inline public/app/plugins/panel/xychart/module.tsx:10]: we should modify Tooltip mode to only support `Single` and `Hidden` (see modifying the `addTooltipOptions` func)

**Review by nmarrs — COMMENTED** [review]
mob review 🤖

File: public/app/plugins/panel/xychart/XYChartPanel.tsx (hunk 2 above)
Comment (Develer) [reply public/app/plugins/panel/xychart/XYChartPanel.tsx:222]: UPlotChart is the only component we are using to render xy chart, that's why it's outside the FF. Old xy chart tooltip exists outside xy chart.

File: public/app/plugins/panel/xychart/XYChartPanel.tsx (hunk 1 above)
Comment (Develer) [reply public/app/plugins/panel/xychart/XYChartPanel.tsx]: just check it and `!important` doesn't make any difference here, so I removed it

**Review by nmarrs — APPROVED** [review]
LGTM :) great job on this!
Screenshot 2023-11-28 at 6 37 00 PM

**Review by leeoniya — APPROVED** [review]
:+1:

**Pull request merged**